restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppMessageHandler.java

```diff
@@ -170,15 +176,24 @@ public void onReceive(Object message) throws Exception {
 
             if (pduAsyncResponse instanceof DefaultPduAsyncResponse && pduAsyncResponse.getResponse() instanceof SubmitSmResp) {
                 SubmitSmResp submitSmResp = (SubmitSmResp) pduAsyncResponse.getResponse();
+                if (logger.isInfoEnabled()) {
+                    logger.info(" ********** SmppMessageHandler received SubmitSmResp: "+submitSmResp +"SubmitSmResp Status:"+submitSmResp.getCommandStatus());
+                }
+
                 String smppMessageId = submitSmResp.getMessageId();
 
                 Object ref = pduAsyncResponse.getRequest().getReferenceObject();
 
                 if (ref != null && ref instanceof Sid) {
                     Sid sid = (Sid) ref;
                     SmsMessage smsMessage = storage.getSmsMessagesDao().getSmsMessage(sid);
-                    //update smppMessageId as well as status to SENT and date sent
-                    smsMessage = smsMessage.setSmppMessageId(smppMessageId).setStatus(SmsMessage.Status.SENT).setDateSent(DateTime.now());
+                    if (submitSmResp.getCommandStatus() != 0) {
```

Comment:
i now this is mainly right, but yesterday i noticed a lot of SMPP contants available in SMPP stack

Reply:
i didtn find any related constant here https://github.com/twitter/cloudhopper-smpp/blob/master/src/main/java/com/cloudhopper/smpp/SmppConstants.java, si its ok

Reply:
is the commandstatus shown in logs somehow? i think it would be interesting to have a log of this that could enable us to correlate err code for a given msg..